src/WebJobs.Script/Workers/IFunctionInvocationDispatcher.cs

```diff
@@ -26,5 +26,7 @@ public interface IFunctionInvocationDispatcher : IDisposable
         Task<bool> RestartWorkerWithInvocationIdAsync(string invocationId);
 
         Task StartWorkerChannel();
+
+        void PreShutdown();
```

Comment:
I welcome name suggestions :-) -- but assume this flow will be changed in the coming months anyway and this is a tactical fix